src/main/java/io/github/antishake/AntiShake.java

```diff
@@ -338,7 +338,8 @@ void setEarliestAccelerometerDataIndex(int earliestAccelerometerDataIndex1) {
    * @return latestAccelerometerDataIndex
    */
   private int getLatestAccelerometerDataIndex() {
-    latestAccelerometerDataIndex = getCircularBuffer().getWritePointer();
+    int wp = getCircularBuffer().getWritePointer();
+    latestAccelerometerDataIndex = wp == 0 ? NO_OF_SAMPLES - 1 : wp - 1;
```

Comment:
Why do we need to change it here. Shouldn't it be sent from Circular buffer? @ruraj